finality-aleph/src/sync/substrate/mod.rs

```diff
@@ -68,11 +68,36 @@ impl<H: SubstrateHeader<Number = BlockNumber>> Header for H {
     }
 }
 
+/// Proper `AlephJustification` or a variant indicating virtual justification
+/// for the genesis block, which is the only block that can be the top finalized
+/// block with no proper justification.
+#[derive(Clone, Debug, Encode, Decode)]
+pub enum InnerJustification {
+    AlephJustification(AlephJustification),
+    Genesis,
+}
+
 /// A justification, including the related header.
 #[derive(Clone, Debug, Encode, Decode)]
 pub struct Justification<H: SubstrateHeader<Number = BlockNumber>> {
```

Comment:
You don't necessarily need this trait bound here, correct? It can be just `H`.

Reply:
I thought it's considered one of the "Rust good practices" to not put type bounds on structs and it's more customary to put them on `impls` and/or functions.